src/evo/providertx.cpp

```diff
@@ -210,7 +270,7 @@ bool CheckProUpRegTx(const CTransaction& tx, const CBlockIndex* pindexPrev, CVal
         // This is a temporary restriction that will be lifted later
         // It is required while we are transitioning from the old MN list to the deterministic list
         Coin coin;
-        if (!GetUTXOCoin(COutPoint(dmn->proTxHash, dmn->nCollateralIndex), coin) || coin.IsSpent()) {
+        if (!coinsView.GetCoin(dmn->collateralOutpoint, coin) || coin.IsSpent()) {
```

Comment:
Can you clarify why `GetUTXOCoin()` is not enough pls?

Reply:
Because all these functions might be called in 2 different situations:
1. When the TX is already mined into a block, in this case `GetUTXOCoin` would be enough
2. When put into the mempool and called from `AcceptToMempool`, in this case `GetUTXOCoin` would only use `pcoinsTip`, which does not include TXs from the mempool.

This was the reason I introducing passing around a reference to CCoinsView

Reply:
Hmm... that's interesting. This imo opens a can of worms if we base our CheckXXX on mempool. There is no guarantee that collateral is going to be mined the same block which in its turn could make proreg*tx invalid _after the fact_. I think we should only allow to reference collaterals that were already mined (DIP3 need another correction I guess). In current system we even require 15 confirmations for MN collaterals before you can start a MN via `mnb` btw.

Reply:
https://github.com/codablock/dash/blob/cb404f758c59b0223abf4eb7c0e3ee7cebe50fe9/src/evo/providertx.cpp#L143 handles this already. Only collaterals with at least one confirmation are allowed when external ones are referenced.